src/packages/hex.gleam

```diff
@@ -3,19 +3,78 @@ import gleam/json
 import gleam/http
 import gleam/result
 import gleam/hackney
+import gleam/list
 import gleam/dynamic.{Decoder} as d
+import time/parse.{parse_iso8601_to_epoch_timestamp}
+import gleam/pgo
+import gleam/string
+import gleam/int
 
 pub type Error {
   HackneyError(hackney.Error)
   JsonError(json.DecodeError)
+  DatabaseError(pgo.QueryError)
+  MiscError(String)
 }
 
-pub fn query() {
+pub fn query(db: pgo.Connection) {
+  try last_scanned = get_last_scanned(db)
+
+  query_all_packages([], 1, last_scanned)
+  |> io.debug
+
+  assert Ok(_) = update_last_scanned(db)
+
+  Ok(1)
+}
+
+const last_scanned_query = "SELECT id, date_trunc('second', scanned_at) FROM previous_hex_api_scan LIMIT 1;"
```

Comment:
The `id` here is never used, it can be removed

Reply:
Couldn't figure out how to decode a tuple of 1 so added than in 😓

Reply:
Something like this:

```gleam
let ints = d.tuple3(d.int, d.int, d.int)
let datetime = d.tuple2(ints, ints)
pgo.execute(a, b, c, d.element(0, datetime_decoder))
```

Reply:
Thanks! All fixed now 😄